examples/density_from_temperature_pressure_humidity.jl

```diff
@@ -0,0 +1,155 @@
+# # Defining a simple parameter set and using it to compute density
+#
+# This script shows how to define a simple parameter set, and then using it to
+# compute density as a function of pressure, temperature, and humidity.
+
+# # Define parameters for computing the density of air
+#
+# First, we build a parameter set suitable for computing the density of
+# _moist_ air --- that is, a mixture of dry air, water vapor, liquid droplets,
+# and ice crystals (the latter two are called "condensates").
+
+using Thermodynamics
+using Thermodynamics.Parameters: AbstractThermodynamicsParameters
+
+struct ConstitutiveParameters{FT} <: AbstractThermodynamicsParameters{FT}
+    gas_constant       :: FT
+    dry_air_molar_mass :: FT
+    water_molar_mass   :: FT
+end
+
+"""
+    ConstitutiveParameters(FT; gas_constant       = 8.3144598,
+                               dry_air_molar_mass = 0.02897,
+                               water_molar_mass   = 0.018015)
+
+Construct a set of parameters that define the density of moist air,
+
+```math
+ρ = p / Rᵐ(q) T,
+```
+
+where ``p`` is pressure, ``T`` is temperature, ``q`` defines the partition
+of total mass into vapor, liqiud, and ice mass fractions, and
+``Rᵐ`` is the effective specific gas constant for the mixture,
+
+```math
+Rᵐ(q) = 
+```
+
+where 
+
+For more information see [reference docs].
+"""
+function ConstitutiveParameters(FT = Float64;
+                                gas_constant       = 8.3144598,
```

Comment:
I had some discussions a while back with SEs asking if the parameters constructors should come with default values. The feedback I got was leaning towards not having the defaults, unless it is really beneficial to the structure of the code. 

From what I understood the defaults should live in a toml file. - I don't have any opinions on this personally. More of a question of convention and how much we want to stick to it in different examples

Reply:
This is a "classroom example" that illustrates some thermodynamic calculations. I think we want to include examples like this in the docs as well to explain the theromdynamic theory that this package implements. This example does not illustrate how to use this package within an Earth system model, necessarily.